go/main.go

```diff
@@ -26,22 +27,41 @@ func main() {
 
 	flag.Parse()
 
-	rand.Seed(time.Now().UnixNano())
-
 	c := &config{
 		runs:        *runs,
 		iters:       *iters,
 		concurrency: *concurrency,
 		ratio:       *ratio,
 		key:         *key,
+		keyRange:    *keyRange,
 		server:      *server,
 		port:        *port,
 		socket:      *socket,
 		protocol:    *protocol,
 		out:         *out,
 	}
-	c.dataStr = strings.Repeat("x", *data)
-	c.dataBytes = []byte(c.dataStr)
+
+	if strings.Contains(*data, "-") {
+		split := strings.Split(*data, "-")
+		if len(split) != 2 {
+			panic(fmt.Errorf("invalid data range: %s", *data))
+		}
+		lower, err := strconv.Atoi(split[0])
+		if err != nil {
+			panic(fmt.Errorf("invalid lower data range: %s", *data))
+		}
+		upper, err := strconv.Atoi(split[1])
+		if err != nil {
+			panic(fmt.Errorf("invalid upper data range: %s", *data))
+		}
+		c.dataRange = []int{lower, upper}
+	} else {
+		d, err := strconv.Atoi(*data)
+		if err != nil {
+			panic(fmt.Errorf("invalid data size: %s", *data))
+		}
+		c.dataRange = []int{d}
```

Comment:
how bout we just store this in dataBytes instead similar to before:
`c.data = []byte(string.Repeat("x", upper))` and then in the tasks we can use lower and upper to reslice c.data? 

also add another check here to only set dataRange if d != 0. makes it easier to check this below